gcsio/src/main/java/com/google/cloud/hadoop/gcsio/PrefixMappedItemCache.java

```diff
@@ -92,6 +92,10 @@ public synchronized GoogleCloudStorageItemInfo getItem(StorageResourceId id) {
    * @return the overwritten item, null if no item was overwritten.
    */
   public synchronized GoogleCloudStorageItemInfo putItem(GoogleCloudStorageItemInfo item) {
+    if (!item.exists()) {
```

Comment:
If we already invalidate cache on creation do we still need this condition?

Reply:
This can actually help assure the correct behavior when call createEmptyObject(s) on gcsStorage object - so that I dont need to override each of these methods.